airflow/utils.py

```diff
@@ -566,6 +633,16 @@ def round_time(dt, delta, start_date=datetime.min):
     >>> round_time(datetime(2015, 9, 13, 0, 0), timedelta(1), datetime(2015, 9, 14, 0, 0))
     datetime.datetime(2015, 9, 14, 0, 0)
     """
+
+    if isinstance(delta, six.string_types):
+        # It's cron based, so it's easy
+        cron = croniter(delta, start_date)
+        prev = cron.get_prev(datetime)
+        if prev == start_date:
+            return start_date
+        else:
+            return prev
+
```

Comment:
@mistercrunch 
I think this code doesn't work.   Shouldn't it be:

```python
    if isinstance(delta, six.string_types):
        cron = croniter(delta, dt)
        return cron.get_prev(datetime)
```

I used this doctest to test for this case:
```
    >>> round_time(datetime(2015, 9, 12, 6, 0), '0 0 * * *', datetime(2015, 9, 2, 0, 0))
    datetime.datetime(2015, 9, 12, 0, 0)
```